protocols/kad/src/behaviour.rs

```diff
@@ -181,10 +181,11 @@ pub struct Config {
     provider_publication_interval: Option<Duration>,
     kbucket_inserts: BucketInserts,
     caching: Caching,
+    protocol_name: String,
```

Comment:
Instead of having a new field, we can make use of the `protocol_config` by using `ProtocolConfig::set_protocol_names`.